src/config/ConfigBuilder.ts

```diff
@@ -93,6 +93,8 @@ export class ConfigBuilder {
                 this.handleCredentials(value);
             } else if (key === 'backupAckToClientEnabled') {
                 this.effectiveConfig.backupAckToClientEnabled = tryGetBoolean(value);
+            } else {
+                console.log(`Warning: Unexpected config key '${key}' is passed to the Hazelcast Client`);
```

Comment:
We don't use `console.log` in the client code but use loggers

Reply:
We cannot use logger since when config is being built there is no logger.

Reply:
Logger depends on config. So config should be built in order a logger to be initialized

Reply:
Then I would throw exceptions on these cases